keps/sig-testing/3041-node-conformance-and-features/README.md

```diff
@@ -0,0 +1,782 @@
+  addition, replace tags like `[NodeAlphaFeature]`.
+
+Clean up labels that means whether special environment is needed:
+
+- Eliminate the `[NodeFeature:]` label by renaming to `[Feature:]` when applicable.
+- Document the meaning of `[NodeConformance]` label.
+- Remove the `[NodeSpecialFeature:]` label.
+- Introduce the `[Environment:Foo]` label to indicate the special environment
+  needed to run the feature.
+
+### Non-Goals
+
+- As `[NodeConformance]` may be a confusing term and may be confused with the
+  k8s Conformance tests, renaming of it is not in a scope of this KEP.
+
+## Proposal
+
+### Clarify the semantic of a `[Feature:]` label
+
+The current [feature definition is](https://github.com/kubernetes/community/blob/32a1c14d04ff78684d78b827ac7c49f70352d509/contributors/devel/sig-testing/e2e-tests.md#kinds-of-tests):
+
+> * `[Feature:.+]`: If a test has non-default requirements to run or targets
+>   some non-core functionality, and thus should not be run as part of the
+>   standard suite, it receives a `[Feature:.+]` label, e.g.
+>   `[Feature:Performance]` or `[Feature:Ingress]`. `[Feature:.+]` tests are not
+>   run in our core suites, instead running in custom suites. If a feature is
+>   experimental or alpha and is not enabled by default due to being incomplete
+>   or potentially subject to breaking changes, it does not block PR merges, and
+>   thus should run in some separate test suites owned by the feature owner(s)
+>   (see Continuous Integration below).
+
+Feature will indicate that things may NOT work on all k8s distros and/or node
+capabilities. This flag will not indicate the Feature Gate state.
+
+Proposed change:
+
+> * `[Feature:.+]`: If a test validates functionality that may only work when
+>   specific node capabilities are enabled or when test depends on the
+>   functionality of underlying components like container runtime, and thus may
+>   need to be skipped on certain environments, it receives a `[Feature:.+]`
+>   label, e.g. `[Feature:AppArmor]`. This label can also be applied to signify that
+>   test validates non-core functionality, like `[Feature:Ingress]`.
+>   `[Feature:.+]` tests are not run in a core suites, but typically can run together
+>   on the standard environment as many capabilities and functionality of underlying
+>   components are pre-enabled on the standard environment. Use `[Environment:Foo]` to
+>   signify the specific environment configuration needed to run the test.
+>   For example if test requires GPU to be configured, but tests the non-optional
+>   feature like a device plugin.
+>   `[Feature:+]` label should not be confused with `[FeatureGate:]` label.
+
+And another two:
+
+> * `[FeatureGate:.+]`: If a test only works when the certain feature gate is
+>   enabled it receives a `[FeatureGate:.+]` label. `[FeatureGate:.+]` tests
+>   must also be marked with the status of this feature gate: `[Alpha]`,
```

Comment:
why do we need the status of the feature gate? the test should work during all the status

Reply:
we have test suites that run only e.g [Alpha] tests, and ones that would want to exclude [Alpha]/[Beta] features

Reply:
ic, thanks

Reply:
Also to know which tests is OK to run for k8s with all optional features turned off. Like if I want all beta features Off